BYOB.md

```diff
@@ -0,0 +1,206 @@
+
+```yaml
+uses: ./../__TOOL_CHECKOUT_DIR__
+id: build
+  with:
+    artifact: ${{ fromJson(inputs.slsa-workflow-inputs).artifact }}
+    content: ${{ fromJson(inputs.slsa-workflow-inputs).content }}
+    username: ${{ fromJson(inputs.slsa-workflow-inputs).username }}
+    password: ${{ inputs.slsa-workflow-secret1 }}
+    token: ${{ inputs.slsa-workflow-secret2 || github.token }}
+```
+
+Note that the `./../__TOOL_CHECKOUT_DIR__` is the path where the TRW repository is checked out by the BYOB framework, so it's accessible locally.
+Notice how we populate the token field: If the user has not passed a value to `inputs.slsa-workflow-secret2`, we default to using the GitHub token `github.token`.
+
+#### Generation of Metadata Layout File
+The last thing to do in the TCA is to [generate the metadata file](https://github.com/laurentsimon/byob-doc/blob/main/internal/callback_action/action.yml#L67-L73) to indicate to the BYOB platform which files to attest to, and which attestations to generate. You can ask the platform to generate several attestations, each attestating to one or more artifacts. The snippet below indicates of a single attestation attesting to a single built artifact `my-artifact`. When the BYOB framework generates the attetation, it will add an extension to it, e.g. `.sigstore` or `.intoto.jsonl` depending on the format used.
+
+```json
+{
+  "version": 1,
+  "attestations": [
+    {
+      "name": "my-artifact",
+      "subjects": [
+        {
+          "name": "my-artifact",
+          "digest": {
+            "sha256": "c71d239df91726fc519c6eb72d318ec65820627232b2f796219e87dcf35d0ab4"
+          }
+        }
+      ]
+    }
+  ]
+}
+```
+
+### Upload Attestations
+In a final ["publish" job](https://github.com/laurentsimon/byob-doc/blob/main/.github/workflows/builder_example_slsa3.yml#L110-L141) of the TRW, we [download the attestations](https://github.com/laurentsimon/byob-doc/blob/main/.github/workflows/builder_example_slsa3.yml#L117C19-L121) and do whatever we'd like with them. In our example, we [upload them as release assets to a GitHub release](https://github.com/laurentsimon/byob-doc/blob/main/.github/workflows/builder_example_slsa3.yml#L123-L141). You may instead upload them to a registry, etc. 
+
+Also think about [returning the attestation to the PW](https://github.com/laurentsimon/byob-doc/blob/main/.github/workflows/builder_example_slsa3.yml#L69-L75) in case end-users want to use the attestations themselves. If you do so, we encourage you to create a [secure-download-attestation(https://github.com/laurentsimon/byob-doc/blob/main/download/attestation/action.yml) Action for your users, e.g. under a download folder in your repository. This will improve user experience as they won't have to be aware of the SLSA repository and its framework.
```

Comment:
```suggestion
Also think about [returning the attestation to the PW](https://github.com/laurentsimon/byob-doc/blob/main/.github/workflows/builder_example_slsa3.yml#L69-L75) in case end-users want to publish the artifacts and attestations themselves. If you do so, we encourage you to create a [secure-download-attestation(https://github.com/laurentsimon/byob-doc/blob/main/download/attestation/action.yml) Action for your users, e.g. under a download folder in your repository. This will improve user experience as they won't have to be aware of the SLSA repository and its framework.
```